mono/utils/mono-error.c

```diff
@@ -106,15 +108,30 @@ void
 mono_error_cleanup (MonoError *oerror)
 {
 	MonoErrorInternal *error = (MonoErrorInternal*)oerror;
-	if (error->error_code == MONO_ERROR_NONE)
+	short int orig_error_code = error->error_code;
+	gboolean free_strings = error->flags & MONO_ERROR_FREE_STRINGS;
+	gboolean has_instance_handle = is_managed_exception (error);
+
+	/* Two cleanups in a row without an intervening init. */
+	g_assert (orig_error_code != MONO_ERROR_CLEANUP_CALLED_SENTINEL);
```

Comment:
This assert is incorrect and would never fail. Found by compiler warning.
Fix is to change `orig_error_code` to unsigned.
https://github.com/mono/mono/pull/11863/files#diff-7959a05a5bd4eae07b3a2204f1d591e9R149